src/braket/experimental/autoqasm/types/types.py

```diff
@@ -41,21 +43,29 @@ def is_qasm_type(val: Any) -> bool:
     return isinstance(val, qasm_types)
 
 
-def qasm_range(start: int, stop: Optional[int] = None, step: Optional[int] = 1) -> oqpy.Range:
-    """Range definition.
-
-    Args:
-        start (int): Start of the range
-        stop (Optional[int]): End of the range. Defaults to None.
-        step (Optional[int]): Step of the range. Defaults to 1.
-
-    Returns:
-        oqpy.Range: oqpy range definition.
-    """
-    if stop is None:
-        stop = start
-        start = 0
-    return oqpy.Range(start, stop, step)
+class Range(oqpy.Range):
+    def __init__(
+        self,
+        start: int,
+        stop: Optional[int] = None,
+        step: Optional[int] = 1,
+        annotations: Optional[Iterable[str | tuple[str, str]]] = None,
+    ):
+        """Creates a range definition.
+
+        Args:
+            start (int): Start of the range.
+            stop (Optional[int]): End of the range. Defaults to None.
+            step (Optional[int]): Step of the range. Defaults to 1.
+            annotations (Optional[Iterable[str | tuple[str, str]]]): Annotations for the range.
+                The annotations can be either a string or a tuple of the form
+                (annotation_name, annotation_value).
```

Comment:
What's this? Why not only simple strings?

Reply:
I would guess this maps to `@annotation_name annotation_value`, but I agree it's confusing. The OpenQASM spec (and AST) only defines annotations as containing a string. If my understanding of this is correct and we want to expose this feature, maybe we can give an example in the doc string.

Reply:
This is a good point, I was blindly copying oqpy behavior here. I'd be open to simplifying this in AutoQASM to just support a list of strings. After all, if someone wants the OQ3 style `@annotation_name annotation_value` they could just pass the string `"annotation_name annotation_value"`.

Reply:
See the OQ3 spec, which implies the name/value structure: https://openqasm.com/language/directives.html#annotations

But I do agree we could just represent this as a single string.

Reply:
+1 I think this is a case where we can think about what feature *we* want to create and then map that to the IR feature.

Reply:
Ok, I've updated this PR so that our API takes annotations as `Optional[str, Iterable[str]]`. It's a little bit of an annoying type hint, but if we only accept `Iterable[str]` we lead the user right into the mistake of:
```
@aq.subroutine(annotations="inline")
```
which, because `str` is itself `Iterable[str]`, would produce
```
@i
@n
@l
@i
@n
@e
```

Reply:
I don't mind the `Optional[str, Iterable[str]]` type. [Job decorator](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/src/braket/jobs/hybrid_job.py#L49) does the same thing for a couple arguments